topdown/http.go

```diff
@@ -337,23 +340,27 @@ func executeHTTPRequest(bctx BuiltinContext, obj ast.Object) (ast.Value, error)
 
 	// Check the system certificates config first so that we
 	// load additional certificated into the correct pool.
-	if tlsUseSystemCerts {
+	if tlsUseSystemCerts != nil && *tlsUseSystemCerts {
```

Comment:
To maybe simplify some of the logic through here (or well, consolidate?) I would recommend adding a helper like:

```golang
func useSystemCerts(<params we might need?>) bool {
    <logic for when to use them>
}
```

The benefit IMO is that we can codify that like logic table in a single spot there, unit test it very thoroughly/easily, and avoid the logic spreading out further below. I would prefer future changes only to have to touch that one spot rather than requiring someone (like me 😄 ) remembering to adjust the `defaultSystemCerts` value. Bonus too is that we don't need to have the new block down at line 388, we'd only have this one place that can add them.

Reply:
👍 this sounds like the right approach. We could take the example above a bit further and have it return the cert pool to use. Then we can unit test that logic to be confident it's correct and also it'll help organize the code nicely.

Reply:
Ope, just saw the second comment - should I update to return the cert pool rather than bool? I like this idea!

Reply:
I think that's the best approach as long as there's some way to compare the cert pools that come back--not sure how easy the Go stdlib makes that.